journal/journal-app.js

```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
-                journalElementTest.setAttribute('widget-id', journalID);
-            
-                journalElementTest.addEventListener('dblclick', function(event) {
-                    document.getElementById('overlay').style.display = 'block';
-                    const currentJournalID = event.target.getAttribute('widget-id');
-                    const journalEntries = document.querySelectorAll('.journal-entry');
-                    hideOtherJournalEntries(journalEntries, currentJournalID);
-                    document.getElementById('modal').style.display = 'block';
-                });
-
-                journalContainer.append(journalElementTest);
+            // Creating journal widget for existing journal
+            createJournalWidget(journalContainer, journalID);
         });
-    }
-    else {
-        journalID = createJournalElement(journalTemplate.id, journalTemplate.title, 
+    } else {
+        journalID = createJournalElement(journalTemplate.id, journalTemplate.title,
             journalTemplate.documentation, journalTemplate.reflection, modalRef);
-            
-            const journalElementTest = document.createElement('div');
-            journalElementTest.classList.add('journal-widget');
-            journalElementTest.textContent = 'Placeholder Title';
-            journalElementTest.setAttribute('widget-id', journalID);
+
+        // Creating journal widget for new journal
+        createJournalWidget(journalContainer, journalID);
+
+        // Save new journal to localStorage
+        journalList.push(journalTemplate);
+        saveNotes(journalList);
+    }
+}
+
+// Extracted function to create and append the journal widget
+function createJournalWidget(journalContainer, journalID) {
+    const journalElementTest = document.createElement('div');
+    journalElementTest.classList.add('journal-widget');
+    // journalElementTest.textContent = 'Insert Title';
+    const journalWidgetTitle = document.createElement('span')
+    journalWidgetTitle.classList.add('journal-widget-title')
+    journalWidgetTitle.textContent = 'Insert Title';
+    journalElementTest.append(journalWidgetTitle);
+    journalElementTest.setAttribute('widget-id', journalID);
+
+    // Create a container for the buttons
+    var buttonContainer = document.createElement('div');
+    buttonContainer.style.position = 'absolute';
+    buttonContainer.style.top = '5px';
+    buttonContainer.style.right = '5px';
+
+    // Create the 'Edit' button
+    var editButton = document.createElement('button');
+    editButton.innerText = 'Edit';
+    editButton.style.fontSize = '12px';
+    editButton.style.padding = '5px 10px';
+    editButton.style.marginRight = '5px';  // Space between buttons
+
+    // Create the 'Delete' button
+    var deleteButton = document.createElement('button');
+    deleteButton.innerText = 'Delete';
+    deleteButton.style.fontSize = '12px';
+    deleteButton.style.padding = '5px 10px';
```

Comment:
Why do we modify the styling of the element here instead of the CSS file?

Reply:
moved to css file